### DISCUSSION TITLE
Suppress warnings threated as errors

### DISCUSSION THREAD

**Discussion by Ronvanl** [zephyrproject-rtos/zephyr#61512] [Q&A]
Hi, 

I have a project where we would like to threat all warnings as errors. However, we would like to exclude the warnings within the zephyr source code. There are several warnings during compilation like unused-parameter and missing-field-initializers caused by zephyr includes. What would be the best approach here? I believe it is OK to ignore/suppress the warnings within the zephyr source but would like to be pedantic about our own code.

The CMakeLists.exe file is as follows: 

```
cmake_minimum_required(VERSION 3.20.0)
find_package(Zephyr REQUIRED HINTS $ENV{ZEPHYR_BASE})
project(app)

target_compile_options(app PRIVATE -Wall -Wextra -Werror)
target_sources(app PRIVATE 
    src/main.cpp)
```

Thanks,
Ron

**Comment by yashi** [comment]
Add `CONFIG_COMPILER_WARNINGS_AS_ERRORS=y`to your `prj.conf`, instead.

BTW, please use triple backticks to encode your code.
````
```
your code here
```
````

**Reply by Ronvanl** [reply]
The solution above is to treat warnings as errors and not the other way around right? The question was the opposite. For example, we have warnings/errors of unused-arguments when including the flash.h when CONFIG_FLASH_EX_OP_ENABLED=n. Or missing-field-initializers when using helper macros from gatt.h (ble). 

The source for these warnings are inside the zephyr header files which we include from zephyr.  I am reading some stuff about including the header files with -isystem but I do not understand how.

**Reply by yashi** [reply]
Ah, sorry.  You wanted to suppress them. (I was confused with the word "threated". would you mind fixing it?)
If you find warnings, please submit issues about them.  I don't see any warnings when I build my apps, though.  Do you care to share how to reproduce?

**Reply by AkselJensen** [reply]
use `target_compile_options(app PRIVATE -Wall -Wextra -Wpedantic -Werror)`, `#include "zephyr/logging/log.h"`
and add `LOG_INFO("you get warnings now");` somewhere. Apparently zephyr headers are not system headers - unless warnings due to macro expansion are not suppressed by "-isystem"?
